src/ucp/tag/tag_recv.c

```diff
@@ -202,6 +209,14 @@ ucp_tag_recv_common(ucp_worker_h worker, void *buffer, size_t count,
         req->recv.tag_mask = tag_mask;
         req->recv.cb       = cb;
         ucp_tag_exp_push(&context->tm, queue, req);
+
+        if (!(req->flags & UCP_REQUEST_FLAG_MATCHED)) {
+            /* If offload supported, post this tag to transport as well.
+             * Check for match flag, because it may be a part of already
+             * matched message in AM protocol (thus we do not need to post
```

Comment:
how can this be ? if offload is enabled, the sender should not send multi-fragment message at all.